### PR TITLE
chore(kv): fixup the checks service to use the index store

### PR DESCRIPTION
**Pull request by jsteenb2**
closes: #16370 

- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass

### CODE REVIEW

File: kv/store_index.go [hunk 1]
```diff
@@ -75,29 +74,97 @@ func (s *IndexStore) DeleteEnt(ctx context.Context, tx Tx, ent Entity) error {
...
-			entVal, err := s.EntStore.FindEnt(ctx, tx, ent)
-			if err != nil {
-				return errors.Wrap(err, "entity lookup")
+	entCaptureFn := opts.CaptureFn
+	kvStream, filterFn := s.indexFilterStream(ctx, tx, opts.FilterEntFn)
+	opts.FilterEntFn = filterFn
+	opts.CaptureFn = func(key []byte, indexVal interface{}) error {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
```
Comment (kelwang) [inline kv/store_index.go]: This error needs to be wrapped.

File: kv/store_base.go [hunk 2]
```diff
@@ -26,7 +26,7 @@ func EncIDKey(ent Entity) ([]byte, string, error) {
 	return id, "ID", err
 }
 
-func EncOrgNameKeyFn(caseSensitive bool) func(ent Entity) ([]byte, string, error) {
+func EncOrgNameKey(caseSensitive bool) func(ent Entity) ([]byte, string, error) {
```
Comment (kelwang) [inline kv/store_base.go]: add comments for exported func

File: kv/store_base.go [hunk 3]
```diff
@@ -40,7 +40,7 @@ func EncBodyJSON(ent Entity) ([]byte, string, error) {
 
 type DecodeBucketEntFn func(key, val []byte) (keyRepeat []byte, decodedVal interface{}, err error)
 
-func DecIndexEntFn(key, val []byte) ([]byte, interface{}, error) {
+func DecIndexEnt(key, val []byte) ([]byte, interface{}, error) {
```
Comment (kelwang) [inline kv/store_base.go]: comments for exported func

File: kv/store_base.go [hunk 4]
```diff
@@ -401,8 +407,12 @@ func (i *Iter) Next(ctx context.Context) (key []byte, val interface{}) {
 	case len(i.prefix) > 0:
 		i.seek(ctx)
 		nextFn = func() ([]byte, []byte) {
-			kv := <-i.seekChan
-			return kv.k, kv.v
+			select {
+			case <-ctx.Done():
+				return nil, nil
```
Comment (kelwang) [inline kv/store_base.go]: not sure why does it return nil here

File: kv/store_base.go [hunk 5]
```diff
@@ -495,6 +501,18 @@ func indexByOrgNameKey(orgID influxdb.ID, name string, caseSensitive bool) ([]by
 	}
 	k := make([]byte, influxdb.IDLength+len(name))
 	copy(k, orgIDEncoded)
+
+	if name == "" {
```
Comment (kelwang) [inline kv/store_base.go]: I would do it before make k

File: kv/store_index.go [hunk 6]
```diff
@@ -75,29 +74,97 @@ func (s *IndexStore) DeleteEnt(ctx context.Context, tx Tx, ent Entity) error {
...
+			if !isValid {
+				close(kvStream)
+			}
+		}()
+		ent, err := s.IndexStore.DecodeToEntFn(key, indexVal)
+		if err != nil {
+			return send(nil, nil, err)
+		}
+
+		entVal, err := s.EntStore.FindEnt(ctx, tx, ent)
```
Comment (kelwang) [inline kv/store_index.go]: this should always be the id, right?

File: kv/store_index.go (hunk 1 above)
Comment (jsteenb2) [reply kv/store_index.go]: the errors get wrapped by the Find cmd in the storage base, extra wrapping here isn't warranted

File: kv/store_base.go (hunk 2 above)
Comment (jsteenb2) [reply kv/store_base.go]: I agree, I'm strugglign with naming convention here. The `EncOrgNameKey` key is a key of the org and name.

File: kv/store_base.go (hunk 4 above)
Comment (jsteenb2) [reply kv/store_base.go]: good question, the `nil, nil` is a signal to the `Iter.isNext` call, it tells it the loop should exit

File: kv/store_base.go (hunk 5 above)
Comment (jsteenb2) [reply kv/store_base.go]: @kelwang, the comment below here explains why this is here. It is to provide a partial key when warranted. Fairly often we lookup entities by `orgID` without a name, this provides that mechanism. When no name is present, it still reponds with an error, but also responds with the key filled with the `orgID`

File: kv/store_index.go (hunk 6 above)
Comment (jsteenb2) [reply kv/store_index.go]: I'm a little lost. Which part is suppose to be the `id`? The entity that you see really just the ID. The abstraction works against the `Entity` type though. What this is doing is a conversion between ID to an entity. This one ends up being that the index val is an ID, but that may not always be the case.

**Comment by desa** [comment]
just tested this branch out and it seems like all of my variables are missing

**Pull request merged**